packages/web3-eth/src/web3_eth.ts

```diff
@@ -249,4 +283,45 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI> {
 			returnFormat,
 		);
 	}
+
+	public async subscribe<T extends keyof RegisteredSubscription>(
+		name: T,
+		args?: ConstructorParameters<RegisteredSubscription[T]>[0],
+	): Promise<InstanceType<RegisteredSubscription[T]>> {
+		const subscription = (await this.subscriptionManager?.subscribe(
+			name,
+			args,
+		)) as InstanceType<RegisteredSubscription[T]>;
+		if (
+			subscription instanceof LogsSubscription &&
+			name === 'logs' &&
+			typeof args === 'object' &&
+			args.fromBlock &&
+			Number.isFinite(Number(args.fromBlock))
+		) {
+			setImmediate(() => {
+				this.getPastLogs({ fromBlock: String(args.fromBlock) })
+					.then(logs => {
+						for (const log of logs) {
+							subscription._processSubscriptionResult(log);
+						}
+					})
+					.catch(e => {
+						subscription._processSubscriptionError(e as Error);
+					});
+			});
+		}
```

Comment:
Can we do this inside `LogsSubscription` implementation? If yes may be a separate issue will be good to refactor it.

Reply:
yes we can move into LogsSubscription, but in this case we need to have duplicate of implementation `getPastLogs` method inside LogsSubscription